packages/blockchain-wallet-v4-frontend/src/scenes/Goals/index.js

```diff
@@ -14,7 +14,7 @@ class ActionsContainer extends React.PureComponent {
     this.state = { error: '' }
   }
 
-  componentWillMount () {
+  componentDidMount () {
     const { payload } = this.props.match.params
```

Comment:
Hey guys, for this file here, should the code in componentDidMount be moved into the constructor? Seeing how this code was originally in componentWillMount, I'm assuming it was initializing a few things.

Reply:
componentDidMount is a specific hook to do all the side effects during initialisation, which is much easier to mock during tests than the constructor

also setState won't trigger update in constructor ( however it could be changed to just setting the state)
and componentDidMount would do synchronous state update before drawing anything